Add a schema for t_status.tbl

hwjsnc opened the pull request:
This commit also adds two new field types:

- 'S' for a signed short. This is required for the difficulty modifiers
  in game_difficulty.

- 'dN' (where N is an integer > 0) for N bytes of data. This is needed
  for a stretch of null bytes in status_p. I also found it useful for
  developing the schema one field at a time.

I have not tested the efficacy fields in status_p, but according to
SoftBrilliant the schema is the same as that of status in t_mons.tbl
and the values seem reasonable.

nnguyen259 commented on schema/status.py:
```diff
@@ -0,0 +1,123 @@
+    "pierce_efficacy" : "s",
+    "strike_efficacy" : "s",
+
+    "zeros_1": "d46",
```
Can you check if all of these are just 0s? If that is not the case, then you might want to compare to the schema of t_mons to check

hwjsnc replied:
Yes, zeros_1 is all zero bytes in every row. The corresponding fields in t_mons are sepith/item drops and some floats, so they're probably unused here. (I didn't check though)

nnguyen259 commented:
Yeah they are very similar between t_status and t_mons. And good work on the 2 new data type!

hwjsnc commented:
By the way, what do you think about the key for "signed short"? 'S' makes sense for shorts, but signed int/byte may be useful for some fields and it would make sense to have a consistent system. 'I' doesn't really communicate "signed int", so perhaps something like '+-s'/'+-i' would be better. 'u2'/'i2' for unsigned/signed short and so on (like numpy does it) would work well, but would also require rewriting the existing schemas. Any suggestions?

nnguyen259 commented:
> By the way, what do you think about the key for "signed short"? 'S' makes sense for shorts, but signed int/byte may be useful for some fields and it would make sense to have a consistent system. 'I' doesn't really communicate "signed int", so perhaps something like '+-s'/'+-i' would be better. 'u2'/'i2' for unsigned/signed short and so on (like numpy does it) would work well, but would also require rewriting the existing schemas. Any suggestions?

Python uses uppercase letter to convey unsigned, but we do it the other way around, so we can either:

- Make it consistent within ourselves by making uppercase letter signed
- Make it also consistent with Python by swapping upper and lowercase

hwjsnc commented:
I'd be fine with either option, really. I slightly prefer leaving it as is just to avoid changing all the exsting schemas though.

nnguyen259 commented:
yeah so we can stick to the current system then. Uppercase for us means signed

hwjsnc committed:
Add a schema for t_status.tbl

This commit also adds two new field types:

- 'S' for a signed short. This is required for the difficulty modifiers
  in game_difficulty.

- 'dN' (where N is an integer > 0) for N bytes of data. This is needed
  for a stretch of null bytes in status_p. I also found it useful for
  developing the schema one field at a time.

I have not tested the efficacy fields in status_p, but according to
SoftBrilliant the schema is the same as that of status in t_mons.tbl
and the values seem reasonable.
